pedpy/methods/acceleration_calculator.py

```diff
@@ -0,0 +1,346 @@
+
+    .. image:: /images/voronoi_density.svg
+        :width: 60 %
+        :align: center
+
+    Args:
+        traj_data (TrajectoryData): trajectory data
+        individual_acceleration (pandas.DataFrame): individual acceleration data from
+            :func:`~acceleration_calculator.compute_individual_acceleration`
+        individual_voronoi_intersection (pandas.DataFrame): intersections of the
+            individual with the measurement area of each pedestrian from
+            :func:`~method_utils.compute_intersecting_polygons`
+        measurement_area (MeasurementArea): area in which the voronoi acceleration
+            should be computed
+
+    Returns:
+        DataFrame containing the columns 'frame' and 'acceleration' in 'm/s^2'
+    """
+    if len(individual_acceleration.index) < len(
+        individual_voronoi_intersection.index
+    ):
+        raise AccelerationError(
+            f"Can not compute the Voronoi acceleration, as the there are less acceleration "
+            f"data (rows={len(individual_acceleration)}) than Voronoi intersection "
+            f"data (rows={len(individual_voronoi_intersection.index)}). This "
+            f"means a person occupies space but has no acceleration at some frames."
+            f"To resolve this either edit your Voronoi intersection data, s.th. "
+            f"it only contains the data that is also contained in the acceleration "
+            f"data. Or use a different acceleration border method when computing the "
+            f"individual acceleration."
+        )
+
+    df_voronoi = pandas.merge(
+        individual_voronoi_intersection,
+        individual_acceleration,
+        on=[ID_COL, FRAME_COL],
+    )
+    df_voronoi[ACC_COL] = (
+        shapely.area(df_voronoi.intersection)
+        * df_voronoi.acceleration
+        / measurement_area.area
+    )
+    df_voronoi_acceleration = df_voronoi.groupby(
+        by=df_voronoi.frame
+    ).acceleration.sum()
+    df_voronoi_acceleration = df_voronoi_acceleration.reindex(
+        list(range(traj_data.data.frame.min(), traj_data.data.frame.max() + 1)),
+        fill_value=0.0,
+    )
+    return pandas.DataFrame(df_voronoi_acceleration)
+
+
+def _compute_individual_acceleration(
+    *,
+    movement_data: pandas.DataFrame,
+    frame_rate: float,
+    movement_direction: Optional[npt.NDArray[np.float64]] = None,
+    compute_acceleration_components: bool = True,
+) -> pandas.DataFrame:
+    """Compute the instantaneous acceleration of each pedestrian.
+
+    Args:
+        movement_data (pandas.DataFrame): movement data
+            (see compute_individual_movement)
+        frame_rate (float): frame rate of the trajectory data
+        movement_direction (np.ndarray): main movement direction on which the
+            actual movement is projected (default: None, when the un-projected
+            movement should be used)
+        compute_acceleration_components (bool): compute the x and y components of the velocity
```

Comment:
"velocity should be replace by "acceleration"

Reply:
done